tests/run/rescue-boundary.scala

```diff
@@ -0,0 +1,76 @@
+import scala.util.control.{ControlException, NonFatal}
+import scala.util.*
+
+object lib:
+  extension [T](op: => T) inline def rescue (fallback: => T) =
+    try op
+    catch
+      case ex: ControlException => throw ex
+      case NonFatal(_) => fallback
```

Comment:
This is a perfect example of why `ControlThrowable`s are *not* caught by `NonFatal`. By making this new `ControlException` by-passing that mechanism, we introduce *more* work for user-space code like this to do the right thing.